enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/SdkSaveQuestionForm.tsx

```diff
@@ -15,18 +17,25 @@ export const SdkSaveQuestionForm = ({ onCancel }: SdkSaveQuestionFormProps) => {
   const { question, originalQuestion, onSave, onCreate, targetCollection } =
     useInteractiveQuestionContext();
 
-  if (!question) {
+  const { id, isLoading } = useValidatedEntityId({
+    type: "collection",
+    id: targetCollection,
+  });
+
+  if (!question || isLoading) {
     return null;
   }
 
+  const finalId = id ?? targetCollection;
+
   return (
     <SaveQuestionProvider
       question={question}
       originalQuestion={originalQuestion ?? null}
       onCreate={onCreate}
       onSave={onSave}
       multiStep={false}
-      targetCollection={targetCollection}
+      targetCollection={isValidId(finalId) ? finalId : undefined}
```

Comment:
the code structure and variable names are quite different from how we do it in `enterprise/frontend/src/embedding-sdk/components/public/CreateDashboardModal/CreateDashboardModal.tsx`

e.g. we don't have `isValidId` there. I'd expect all places that resolves collection ID to look somewhat similar, not with such difference.